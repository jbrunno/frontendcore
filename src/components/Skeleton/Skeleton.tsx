import React from 'react';
import {
  Skeleton as MuiSkeleton,
  SkeletonProps as MuiSkeletonProps,
} from '@mui/material';

export type SkeletonProps = MuiSkeletonProps;

export function Skeleton(props: SkeletonProps) {
  return <MuiSkeleton {...props} />;
}

export default Skeleton;
