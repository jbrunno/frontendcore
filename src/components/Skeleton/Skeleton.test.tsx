import React from 'react';
import { render } from '@test-utils/component-testing';
import { Skeleton } from './Skeleton';

test('renders the Skeleton', () => {
  const { container } = render(
    <Skeleton variant="circular" animation="pulse" />,
  );
  expect(container).toMatchSnapshot();
});
