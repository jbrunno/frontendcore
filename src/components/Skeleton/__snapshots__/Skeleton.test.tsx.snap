// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Skeleton 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  display: block;
  background-color: rgba(59, 59, 65, 0.11);
  height: 1.2em;
  border-radius: 50%;
  -webkit-animation: animation-0 1.5s ease-in-out 0.5s infinite;
  animation: animation-0 1.5s ease-in-out 0.5s infinite;
}

<div>
  <span
    class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse emotion-0"
  />
</div>
`;
